agent/functional_tests/tests/functionaltests_test.go

```diff
@@ -539,6 +539,82 @@ func telemetryTest(t *testing.T, taskDefinition string) {
 	assert.NoError(t, err, "Task stopped, verify metrics for memory utilization failed")
 }
 
+func telemetryTestWithStatsPolling(t *testing.T, taskDefinition string) {
+	// Try to use a new cluster for this test, ensure no other task metrics for this cluster
+	newClusterName := "ecstest-telemetry-polling-" + uuid.New()
+	_, err := ECS.CreateCluster(&ecsapi.CreateClusterInput{
+		ClusterName: aws.String(newClusterName),
+	})
+	require.NoError(t, err, "Failed to create cluster")
+	defer DeleteCluster(t, newClusterName)
+
+	// additional config fields to use polling instead of stream
+	agentOptions := AgentOptions{
+		ExtraEnvironment: map[string]string{
+			"ECS_CLUSTER":                       newClusterName,
+			"ECS_POLL_METRICS":                  "true",
+			"ECS_POLLING_METRICS_WAIT_DURATION": "15s",
+		},
+	}
+	agent := RunAgent(t, &agentOptions)
+	defer agent.Cleanup()
+
+	params := &cloudwatch.GetMetricStatisticsInput{
+		MetricName: aws.String("CPUUtilization"),
+		Namespace:  aws.String("AWS/ECS"),
+		Period:     aws.Int64(60),
+		Statistics: []*string{
+			aws.String("Average"),
+			aws.String("SampleCount"),
+		},
+		Dimensions: []*cloudwatch.Dimension{
+			{
+				Name:  aws.String("ClusterName"),
+				Value: aws.String(newClusterName),
+			},
+		},
+	}
+	params.StartTime = aws.Time(RoundTimeUp(time.Now(), time.Minute).UTC())
+	params.EndTime = aws.Time((*params.StartTime).Add(waitMetricsInCloudwatchDuration).UTC())
```

Comment:
i don't think the params here are used?

Reply:
Good catch! I shortened this test's checks and didn't remove the set up part. Removed.